gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -25,5 +34,602 @@
+    }
+  }
+
+  /**
+   * Class for specifying table get, create and update options.
+   */
+  class TableOption extends Option {
+
+    private static final long serialVersionUID = -1723870134095936772L;
+
+    private TableOption(BigQueryRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the table's fields to be returned by the RPC call. If this
+     * option is not provided all table's fields are returned. {@code TableOption.fields}) can be
+     * used to specify only the fields of interest. {@link BaseTableInfo#tableId()} and
+     * {@link BaseTableInfo#type()} are always returned, even if not specified.
+     */
+    public static TableOption fields(TableField... fields) {
+      return new TableOption(BigQueryRpc.Option.FIELDS, TableField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying table data list options.
+   */
+  class TableDataListOption extends Option {
+
+    private static final long serialVersionUID = 8488823381738864434L;
+
+    private TableDataListOption(BigQueryRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the maximum number of rows to be returned.
+     */
+    public static TableDataListOption maxResults(long maxResults) {
+      return new TableDataListOption(BigQueryRpc.Option.MAX_RESULTS, maxResults);
+    }
+
+    /**
+     * Returns an option to specify the page token from which to start listing table data.
+     */
+    public static TableDataListOption startPageToken(String pageToken) {
+      return new TableDataListOption(BigQueryRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option that sets the zero-based index of the row from which to start listing table
+     * data.
+     */
+    public static TableDataListOption startIndex(long index) {
+      return new TableDataListOption(BigQueryRpc.Option.START_INDEX, index);
+    }
+  }
+
+  /**
+   * Class for specifying job list options.
+   */
+  class JobListOption extends Option {
+
+    private static final long serialVersionUID = -8207122131226481423L;
+
+    private JobListOption(BigQueryRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to list all jobs, even the ones issued by other users.
+     */
+    public static JobListOption allUsers() {
+      return new JobListOption(BigQueryRpc.Option.ALL_USERS, true);
+    }
+
+    /**
+     * Returns an option to list only jobs that match the provided filters.
+     */
+    public static JobListOption stateFilter(JobStatus.State... stateFilters) {
+      List<String> stringFilters = Lists.transform(ImmutableList.copyOf(stateFilters),
+          new Function<JobStatus.State, String>() {
+            @Override
+            public String apply(JobStatus.State state) {
+              return state.toString().toLowerCase();
```

Comment:
nit, `name()` is slightly better than `toString()` as the latter could be overridden.